src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 1 << 4;
+    private Node<K, V>[] table;
+    private int capacity;
+    private int size;
+    private int threhold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threhold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size == threhold) {
+            resize();
+        }
+        int bucket = key == null ? 0 : Math.abs(key.hashCode() % capacity);
```

Comment:
lets move this logic to separate private method as we use it several times in code